### PR TITLE
chore(ts): refactor types into separate modules

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #941 

## Description of the change:

Move common types into `types.ts` to avoid cirular deps. Also, fix some type annotations that eslint might have missed.

## Motivation for the change:

See #941

### CODE REVIEW

**Comment by tthvo** [comment]
@andrewazores I just realized this is quite a huge set of changes. Do you think we have time to review for 2.4? 

If not, I think we could delay this till next release and use it as a chance to go over web UI functionalities from top to toe?

**Comment by andrewazores** [comment]
I would rather try to get this in ASAP since it is so large and rebasing it will only get harder. It may not make it in time for the 2.4 release but that's fine, it can be merged into main after the release window as soon as it is ready as one of the first merges into the next development cycle.

**Comment by tthvo** [comment]
Suree just some little moving around and lots of tests broken left. Will try to get it done asap.

**Comment by tthvo** [comment]
Ready now ^^ Most changes are just moving definitions into new files. The following are the conventions:

- `.types.ts(x)`: Define type definitions. This includes types and enums.
- `.utils.ts(x)`: Define utility functions. Also, might contain constants (usually tightly coupled with the utility functions).
- `.const.ts`: Define constants. These are mostly constants that are purely for UI rendering.
- `context.tsx`: Define React contexts. Mostly, to resolve circular deps. These can be defined in `utils.ts(x)`.

I tried to move definitions closed to where they are used (colocation?). Quite a huge change but I guess its a good chance to go over the web UI as a whole.

**Comment by andrewazores** [comment]
> Ready now ^^ Most changes are just moving definitions into new files. The following are the conventions:
> 
>     * `.types.ts(x)`: Define type definitions. This includes types and enums.
> 
>     * `.utils.ts(x)`: Define utility functions. Also, might contain constants (usually tightly coupled with the utility functions).
> 
>     * `.const.ts`: Define constants. These are mostly constants that are purely for UI rendering.
> 
>     * `context.tsx`: Define React contexts. Mostly, to resolve circular deps. These can be defined in `utils.ts(x)`.
> 
> 
> I tried to move definitions closed to where they are used (colocation?). Quite a huge change but I guess its a good chance to go over the web UI as a whole.

Document these conventions in README or a CONTRIBUTING doc please :-)

File: src/app/Dashboard/cryostat-dashboard-templates.tsx
Comment (andrewazores) [inline src/app/Dashboard/cryostat-dashboard-templates.tsx:18]: May as well bump this now since 2.4 is imminent.

File: src/app/RecordingMetadata/ClickableLabel.tsx
Comment (andrewazores) [inline src/app/RecordingMetadata/ClickableLabel.tsx:38]: Maybe these things can also be extracted to consts?

File: src/app/Shared/Services/api.types.ts
Comment (andrewazores) [inline src/app/Shared/Services/api.types.ts:20]: This should have `v2.3` as well. There are no new `v2.4` API endpoints but it could be added here as well for consistency.

File: src/app/Shared/Services/api.types.ts
Comment (tthvo) [reply src/app/Shared/Services/api.types.ts:92]: Right sounds good!

File: src/app/RecordingMetadata/ClickableLabel.tsx
Comment (tthvo) [reply src/app/RecordingMetadata/ClickableLabel.tsx:38]: Would u prefer it to be a separate issue/PR to clean up the overall css?

Comment (tthvo) [reply src/app/RecordingMetadata/ClickableLabel.tsx:38]: https://github.com/cryostatio/cryostat-web/issues/853

File: src/app/Shared/Services/api.types.ts
Comment (tthvo) [reply src/app/Shared/Services/api.types.ts:92]: Actually, that would mean we need to split out the `api.utils.ts`? Might cluster the directory?

Comment (andrewazores) [reply src/app/Shared/Services/api.types.ts:92]: Hmm... okay, let's leave it as it is for now then.

Comment (tthvo) [reply src/app/Shared/Services/api.types.ts:92]: Okayy!!

**Comment by andrewazores** [comment]
Sorry @tthvo ... this needs a big rebase after #1117

**Comment by tthvo** [comment]
Ah right! I will go over it by this week!

**Comment by tthvo** [comment]
Phew, finished rebasing :)) Also, added a small padding for recording option label group.

| Before | After |
|:---:|:---:|
|before |after|

**Review by andrewazores — APPROVED** [review]
Awesome work, thanks @tthvo

**Comment by andrewazores** [comment]
I do see one last dependency cycle:

```
$ yarn run lint
[format] [prettier:check] Checking formatting...
[format] [license:check] No default format specified. Using {"prepend":"/*","append":"*/"} as backup
[format] [license:check] ✔ All files have licenses.
[format] [license:check] Command succeeded
[format] [license:check] yarn run license:check exited with code 0
[format] [prettier:check] All matched files use Prettier code style!
[format] [prettier:check] yarn run prettier:check exited with code 0
[format] yarn run format exited with code 0
[type-check] yarn run type-check exited with code 0
[eslint] 
[eslint] /home/work/workspace/cryostat-web/src/app/Shared/Redux/ReduxStore.tsx
[eslint]   23:1  warning  Dependency cycle detected  import/no-cycle
[eslint] 
[eslint] ✖ 1 problem (0 errors, 1 warning)
[eslint] 
[eslint] yarn run eslint exited with code 0

```

**Comment by tthvo** [comment]
Hmm, not seeing it on my end tho. Any idea?

```bash
$ yarn lint
[eslint] yarn run eslint exited with code 0
[format] [prettier:check] Checking formatting...
[format] [license:check] No default format specified. Using {"prepend":"/*","append":"*/"} as backup
[format] [license:check] ✔ All files have licenses.
[format] [license:check] Command succeeded
[format] [license:check] yarn run license:check exited with code 0
[format] [prettier:check] All matched files use Prettier code style!
[format] [prettier:check] yarn run prettier:check exited with code 0
[format] yarn run format exited with code 0
[type-check] yarn run type-check exited with code 0
```

**Comment by tthvo** [comment]
Just double checked. That should be fixed with `import type { RootState } from '../ReduxStore';` within `src/app/Shared/Redux/Middlewares/PersistMiddleware.tsx`

There is no complaints with github either?

image

**Comment by andrewazores** [comment]
Weird. Good enough for me - I'll figure out what's happening on my setup then.

**Pull request merged**
